### PR TITLE
Feature/rfs 83 add event logs for commit federation

### PR DESCRIPTION
**Pull request by martinmedina**
## Log added for commit federation event
There is a need to know when a new federation is committed.

### Data to include in event log
- Old federation address 
- Old federation public keys
- New federation address
- New federation public keys
- Block number when new federation will be activated

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 138 issues
* ![MAJOR][MAJOR] 99 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 32 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L277](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L277): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L306](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L306): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L317](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L317): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L322](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L322): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L339](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L339): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L343](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L343): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L352](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L352): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L360](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L360): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L383](https://github.com/rsksmart/rskj/blob/87ee4b9a254d131d7b4a832ed6d4db4e635aed2d/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L383): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    public void logUpdateCollections(Transaction rskTx) {
+        this.logs.add(
+                new LogInfo(BRIDGE_CONTRACT_ADDRESS,
+                            Collections.singletonList(Bridge.UPDATE_COLLECTIONS_TOPIC),
+                            RLP.encodeElement(rskTx.getSender())
+                )
+        );
+    }
+
+    public void losAddSignature(BtcECKey federatorPublicKey, BtcTransaction btcTx, byte[] rskTxHash) {
```
Comment (aeidelman) [inline rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java]: Looks like there is a typo in this method name

Comment (martinmedina) [reply rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java]: My fingers bad, fixed!

**Comment by rskops** [comment]
SonarQube analysis reported 138 issues
* ![MAJOR][MAJOR] 99 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 32 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L277](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L277): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L306](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L306): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L317](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L317): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L322](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L322): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L339](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L339): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L343](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L343): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L352](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L352): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L360](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L360): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L383](https://github.com/rsksmart/rskj/blob/d8ca420b718330258b78327b4ebe611527b3d412/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L383): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.stream.Collectors;
+
+/**
+ * Responsible for logging events triggered by BridgeContract.
+ *
+ * @author martin.medina
+ */
+public class BridgeEventLoggerImpl implements BridgeEventLogger {
+
+    private static final byte[] BRIDGE_CONTRACT_ADDRESS = TypeConverter.stringToByteArray(PrecompiledContracts.BRIDGE_ADDR);
```
Comment (amendelzon) [inline rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java]: Maybe better if this address gets passed in as the Bridge also gets its address passed in.

**Pull request merged**